File: zig/collatz-conjecture/collatz_conjecture.zig
// Please implement the `ComputationError.IllegalArgument` error.

pub const ComputationError = error{IllegalArgument};

pub fn steps(number: usize) anyerror!usize {
    if (number <= 0) {
        return ComputationError.IllegalArgument;
    }
    var num = number;
    var count: usize = 0;
    while (num != 1) {
        if (num % 2 == 0) {
            num /= 2;
        } else {
            num = (num * 3) + 1;
        }
        count += 1;
    }
    return count;
}

File: c/collatz-conjecture/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
  int count = 0;
  if (start < 1) {
    return -1;
  }
  while (start != 1) {
    count += 1;
    if ((start % 2) == 0) {
      start /= 2;
    } else {
      start = (3 * start) + 1;
    }
  }
  return count;
}
